src/domain/datasets/services/src/utils/create_dataset_use_case_helper.rs

```diff
@@ -185,3 +181,27 @@ impl CreateDatasetUseCaseHelper {
 }
 
 ////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
+
+#[derive(Error, Debug)]
+pub enum CanonicalDatasetAliasError {
+    #[error(transparent)]
+    IncorrectAliasAccountName(#[from] IncorrectAliasAccountNameError),
```

Comment:
I disagree a bit with this error type. I don't think that user `alice` trying to create a dataset under `bob/foo` is an "incorrect alias error" - if that was the case - our API should not have accepted a multi-tenant `DatasetAlias` to begin with and only accepted `DatasetName`.

I definitely see in future cases where `alice` creates a dataset under another account, e.g. an organization that she is a part of. So in this case a more natural errors to me are:
- **unauthorized** - you don't have permissions to create a dataset under this different users
- **not found** - the account you're referring to does not exist (or you don't have permissions to see it)

Reply:
While we don't have organizations - I would just return unauthorized for all cases

Reply:
> I definitely see in future cases where alice creates a dataset under another account, e.g. an organization that she is a part of

A fair comment